python/ray/data/read_api.py

```diff
@@ -1235,6 +1248,85 @@ def read_binary_files(
     )
 
 
+@PublicAPI(stability="alpha")
+def read_sql(
+    sql: str,
+    connection_factory: Callable[[], Connection],
+    *,
+    parallelism: int = -1,
+    ray_remote_args: Optional[Dict[str, Any]] = None,
+):
+    """Read from a database that provides a
+    `Python DB API2-compliant <https://peps.python.org/pep-0249/>`_ connector.
+
+    .. note::
+
+        By default, ``read_sql`` launches multiple read tasks, and each task executes a
+        ``LIMIT`` and ``OFFSET`` to fetch a subset of the rows. However, for many
+        databases, ``OFFSET`` is slow.
+
+        As a workaround, you can set ``parallelism=1`` to directly fetch all rows.
```

Comment:
Let's be more explicit:

```
As a workaround, you can set ``parallelism=1`` to directly fetch all rows in one task. But it requires the size of all result rows can fit in memory of single task, otherwise it will cause OOM.
```